### PR TITLE
feat(es/parser, es/codegen, es/visit): support  class static blocks

### PR DESCRIPTION
**Pull request by sosukesuzuki**
- [x] AST definition
- [x] Visitor support
- [x] Parser support (ES)
- [x] Parser support (TS)
- [x] Codegen

~~- [ ] Transformer support~~

### CODE REVIEW

File: ecmascript/parser/src/lib.rs [hunk 1]
```diff
@@ -295,6 +302,10 @@ pub struct TsConfig {
     /// Stage 3.
     #[serde(default)]
     pub import_assertions: bool,
+
+    #[serde(default)]
+    #[serde(rename = "staticBlocks")]
+    pub static_blocks: bool,
```
Comment (kdy1) [inline ecmascript/parser/src/lib.rs:308]: Is static block optional for typescript?
If `tsc` allows it by default, `swc` should allow it without a configuration, and if `tsc` does not support it, `swc` should not allow it for typescript files.

Comment (sosukesuzuki) [reply ecmascript/parser/src/lib.rs:308]: Since TS 4.4, it is not optional, fully supported. I'll fix it.

Comment (kdy1) [reply ecmascript/parser/src/lib.rs:308]: Thanks!

**Comment by kdy1** [comment]
How do you think about splitting this into parser / codegen PR (done?) and transform PR?
It will unblock Deno.

cc @dsherret
(https://github.com/swc-project/swc/issues/2125#issuecomment-907667223)

**Comment by sosukesuzuki** [comment]
It sounds good.

File: ecmascript/codegen/src/lib.rs [hunk 2]
```diff
@@ -1216,6 +1217,20 @@ impl<'a> Emitter<'a> {
         }
     }
 
+    #[emitter]
+    fn emit_static_block(&mut self, n: &StaticBlock) -> Result {
+        self.emit_leading_comments_of_span(n.span(), false)?;
+        keyword!("static");
+        punct!("{");
```
Comment (kdy1) [inline ecmascript/codegen/src/lib.rs:1224]: Maybe `formatting_space!()` here?
It's not important, though.

**Review by kdy1 — APPROVED** [review]
I'm almost fine with it.
Are you okay with merging the PR?

File: ecmascript/codegen/src/lib.rs (hunk 2 above)
Comment (sosukesuzuki) [reply ecmascript/codegen/src/lib.rs:1225]: `f3e295f` :+1:

**Comment by sosukesuzuki** [comment]
@kdy1 I'm okay with merging this PR! But please review some additional commits

**Review by kdy1 — APPROVED** [review]
Additional commits seem harmless to me.

**Pull request merged**